### PR TITLE
Feature/check for hard coded strings

### PR DESCRIPTION
**Pull request by norkator**
* Mostly suppressing classes with HardCodedStringLiteral where only string parameter names etc.
* Suppress more carefully some classes having lot of actual strings.
* Found one or two string to extract.

### CODE REVIEW

File: app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java [hunk 1]
```diff
@@ -227,19 +232,6 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        // Install security provider, which is not supported by android kitkat for ssl connections
-        if (Build.VERSION.SDK_INT == Build.VERSION_CODES.KITKAT) {
-            try {
-                ProviderInstaller.installIfNeeded(getApplicationContext());
-            } catch (GooglePlayServicesRepairableException e) {
-                e.printStackTrace();
-            } catch (GooglePlayServicesNotAvailableException e) {
-                e.printStackTrace();
-            }
-        }
```
Comment (ristomatti) [inline app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java:239]: Should this have been removed? I can't find a related commit.

File: app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java [hunk 2]
```diff
@@ -254,29 +246,14 @@ protected void onCreate(Bundle savedInstanceState) {
...
         }
-        /*if (!SP_PARCELS_AUTOMATIC_UPDATE) {
-            if (isMyServiceRunning(ParcelServiceTimer.class)) {
-                stopService(new Intent(MainMenu.this, ParcelServiceTimer.class));
-            }
-        }*/
 
 
         // Intro logic, show intro if not shown yet
-        if (!sharedPreferences.getBoolean(Constants.SP_INTRO_IS_SHOWN, false) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```
Comment (ristomatti) [inline app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java:265]: How is this related?

File: app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java [hunk 3]
```diff
@@ -254,29 +246,14 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
-        // -----------------------------------------------------------------------------------------
-
-        // Status bar tint
-        if (Build.VERSION.SDK_INT == Build.VERSION_CODES.KITKAT) {
-            setTranslucentStatus(true);
-            SystemBarTintManager tintManager = new SystemBarTintManager(this);
-            tintManager.setStatusBarTintEnabled(true);
-            tintManager.setStatusBarTintResource(R.color.colorPrimary);
-        }
```
Comment (ristomatti) [inline app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java:278]: How is this related?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -285,6 +285,7 @@
     <string name="no_parcels">No parcels added</string>
     <string name="guide_part_one">Press</string>
     <string name="guide_part_two">to add</string>
+    <string name="reference_to_database_is_null_cannot_add_items">Reference to database is null, cannot add items!</string>
```
Comment (ristomatti) [inline app/src/main/res/values/strings.xml:288]: "is null" -> "is missing"

File: app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java (hunk 2 above)
Comment (norkator) [reply app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java:265]: Sorry again my ugly habbit, I was looking while doing this pr related thing that there's code which is not valid anymore due min sdk verison. 😭

File: app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java (hunk 3 above)
Comment (norkator) [reply app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java:278]: 😭  again same, not related at all!

File: app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java (hunk 1 above)
Comment (norkator) [reply app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java:239]: Not in this branch for sure. 😭

**Comment by norkator** [comment]
> two ugly merge commits

So wait... I modified Jenkins file directly at master and wanted it to this pr, I did update from master, that is bad habbit?

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (norkator) [reply app/src/main/res/values/strings.xml]: https://github.com/norkator/paketin-seuranta/pull/32/commits/822733aae89e1a68b4778c0704c5b04009ae428b

File: app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java (hunk 3 above)
Comment (ristomatti) [reply app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java:278]: Well what can you do...

File: app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java (hunk 2 above)
Comment (ristomatti) [reply app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java:265]: It happens to everyone occasionally. The temptation is too high. :slightly_smiling_face:

File: app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java (hunk 1 above)
Comment (ristomatti) [reply app/src/main/java/com/nitramite/paketinseuranta/MainMenu.java:239]: I guess you're not planning to revert it from this branch so I'll just approve. :slightly_smiling_face:
